grpc/shared/bech32m.go

```diff
@@ -0,0 +1,92 @@
+package shared
+
+// Copied from astria-cli-go bech32m module (https://github.com/astriaorg/astria-cli-go/blob/d5ef82f718325b2907634c108d42b503211c20e6/modules/bech32m/bech32m.go#L1)
+// TODO: organize the bech32m usage throughout the codebase
+
+import (
+	"crypto/ed25519"
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcutil/bech32"
+)
+
+type Address struct {
+	address string
+	prefix  string
+	bytes   [20]byte
+}
+
+// String returns the bech32m address as a string
+func (a *Address) String() string {
+	return a.address
+}
+
+// Prefix returns the prefix of the bech32m address
+func (a *Address) Prefix() string {
+	return a.prefix
+}
+
+// Bytes returns the underlying bytes for the bech32m address as a [20]byte array
+func (a *Address) Bytes() [20]byte {
+	return a.bytes
+}
+
+// ValidateBech32mAddress verifies that a string in a valid bech32m address. It
+// will return nil if the address is valid, otherwise it will return an error.
+func ValidateBech32mAddress(address string, intendedPrefix string) error {
```

Comment:
do these need to be exported?

Reply:
yes, because the execution server is also using. https://github.com/astriaorg/flame/blob/211b66d664881cfe0a0a859c17d1babb87b9403b/grpc/execution/server.go#L226 we use it to validate the bech32 address when we update it with the block number.

Reply:
Should we just validate all of the addresses on parse of the config or in sharedcontainer.

Admittedly this logic will need updating with newest geth changes, so probably best not to over optimize right now.